packages/test/test-end-to-end-tests/src/test/gc/gcSweepDataStores.spec.ts

```diff
@@ -173,18 +169,36 @@ describeCompat("GC data store sweep tests", "NoCompat", (getTestObjectProvider)
 		};
 	};
 
-	const setupContainerCloseErrorValidation = (container: IContainer) => {
-		container.on("closed", (error) => {
-			assert(error !== undefined, `Expecting an error!`);
-			assert(error.message.startsWith("DataStore was deleted:"));
-		});
+	/**
+	 * Returns a promise that will resolve when all the passed in containers have closed dur
+	 * to "DataStore was deleted" error.
+	 */
+	const getContainersClosePromise = async (containers: IContainer[]) => {
+		const closePromises: Promise<void>[] = [];
+		for (const container of containers) {
+			const promise = timeoutPromise((resolve) => {
+				const listener = (error: IErrorBase | undefined) => {
+					assert(error !== undefined, `Expecting an error!`);
```

Comment:
The thrown errors for these asserts will go to the code that triggered the "closed" event, is that what you want?  And then the `container.off` call will be missed, and the Promise will not resolve (nor will it reject).  Maybe this is where timeoutPromise helps?

Reply:
Good catch. That was totally unintentional, I thought the promise would reject. I changed it to reject instead.